src/components/KeywordGrab.js

```diff
@@ -33,27 +65,21 @@ const KeywordGrab = () => {
             // Using fetch to fetch the api from
             // flask server it will be redirected to proxy
             let url = '';
-            if (cookies["groupCode"] == 0)
+            if (cookies["groupCode"] != 0)
             {
                 updateGroupMember(cookies['groupCode'], 'keywords', keywordRef.current.value);
-                url="http://localhost:5000/data"
-                setURL("http://localhost:5000/data")
             }
-            else
+            
+            if (cookies['groupCode'] != 0 && cookies["host"] != 0)
             {
-                if(cookies["host"] == 0)
-                {
-                    url="http://localhost:5000/groupMode"
-                    setURL("http://localhost:5000/groupMode")
-                }
-                else
-                {
-                    url="http://localhost:5000/groupMode"
-                    setURL("http://localhost:5000/groupMode")
-                }
+                navigate("/hostRoom")
+            }
+            if (cookies["groupCode"] != 0)
+            {
+                setURL("http://localhost:5000/groupMode")
             }
 
-            fetch(urlString || url, {
+            fetch("http://localhost:5000/data", {
```

Comment:
This doesn't account for group quizzes. Shouldn't the url go to /groupMode in those cases?